test/translation.js

```diff
@@ -1,60 +1,160 @@
+  const buf = new Buffer(hash_hex.replace(/^0x/, '1220'), 'hex')
+  return bs58.encode(buf)
+}
+
 contract("TranslationContract", accounts => {
   const [joshs_address, marys_address, franks_address] = accounts;
   const Languages = { English: 1, Spanish: 2, Chinese: 3, French: 4 };
 
   it("should return the string to be translated", async () => {
     const translationInstance = await TranslationContract.deployed();
+    
+    const newTranslation = async (obj) => {
+
+      //Adding string to IPFS 
+      ipfs.add(new Buffer(obj.string), (err, result) => {
+          if (err) {
+              console.error("Content submission error:", err);
+              return false;
+          } else if (result && result[0] && result[0].hash) {
+              console.log("Successful request. IPFS address:", result[0].hash);
+              //make contract calls to store the IPFS hash in Bytes32
+              translationInstance.newTranslation(
+                ipfsHashToBytes32(result[0].hash),
+                obj.from,
+                obj.to,
+                obj.data
+              );
+              console.log( ipfsHashToBytes32(result[0].hash));
+
+          } else {
+              console.log(result);
+              console.error("Unresolved content submission error");
+              return null;
+          }
+      });
+    }
+
+    const firstRequest =  {
+      string: "asdfa", 
+      from: Languages.English, 
+      to: Languages.French, 
+      data: {from: joshs_address, value: 10 }};
 
-    await translationInstance.newTranslation(
-      "Test Translation",
-      Languages.English,
-      Languages.French,
-      {from: joshs_address, value: 10 }
-    );
-
-    await translationInstance.newTranslation(
-      "Hi, how are you?",
-      Languages.English,
-      Languages.French,
-      {from: joshs_address, value: 10 }
-    );
+    await newTranslation(firstRequest);
+
+    const secondRequest = {
+      string: "Hi, how are you?",
+      from: Languages.English,
+      to: Languages.French,
+      data: {from: joshs_address, value: 10 }};
     
-    await translationInstance.newTranslation(
-      "Third",
-      Languages.English,
-      Languages.French,
-      {from: joshs_address, value: 10 }
-    );
+    await newTranslation(secondRequest);
 
     const arrayOfIds = await translationInstance.getAllOpenRequests.call();
-    const str = await translationInstance.getRequestString(arrayOfIds[1]);
+    const requestHash = await translationInstance.getRequestHash(arrayOfIds[1]);
 
-    // const str = toAscii(result[1]).replace(/\0/g, "");
+    console.log("trying to get my string back. Hash is: " + requestHash)
+   
+    //convert to form consumable by IPFS
+    const bytes32toIPFS = bytes32ToIPFSHash(requestHash);
 
-    console.log("Translation String = ", str);
-    assert.equal(str, "Hi, how are you?");
+    //grab the request string from IPFS
+   await ipfs.cat(bytes32toIPFS, {buffer:true}).then(function(res){
```

Comment:
Spacing? Make sure to use linting + [Prettier](https://github.com/prettier/prettier) so this stuff is automatically taken care of. I understand that linting is not setup for this part of the project though.